src/ui/hooks/comments/useAddComment.tsx

```diff
@@ -59,6 +59,12 @@ export default function useAddComment() {
           query: GET_USER_ID,
         });
 
+        // check if the comment has already been added to the cache
+        // due to comments getting polled in parallel
+        if (data.recording.comments.some((c: any) => c.id === commentId)) {
```

Comment:
My understanding was that this would still lead to duplicated comments. Order of events being:

1) Optimistic update data for `newComment1` is added to the cached query.
2) UI updates with the optimistic data.
3) Real server response data for `newComment1` comes back.
4) The optimistic update data is still in the cached query (due to polling being in flight).
5) This check compares the IDs for the optimistic and real `newComment1`, but the optimistic data has a different ID (`new Date().toISOString()`) compared to the real data (UUID generated by the server).
6) The condition doesn't pass and the function doesn't return early.
7) The real server response data is added to the cached query.
8) UI updates ends up with a duplicate comment, with both the optimistic and real data.

Am I missing something?

Reply:
> My understanding was that this would still lead to duplicated comments.

That's what I would have expected as well, but when I reproduced the issue I saw that Apollo's cache works differently than I expected:

> The optimistic update data is still in the cached query

Surprisingly it isn't when I reproduce this. I don't know why.